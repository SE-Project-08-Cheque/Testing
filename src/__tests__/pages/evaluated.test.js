import React from 'react'
import {render, screen , waitFor} from '@testing-library/react'

import Evaluated from '../../pages/evaluated'
import { BrowserRouter } from "react-router-dom";

import {mocked} from 'ts-jest/utils'
import renderer from 'react-test-renderer'

import API_Service from '../../Services/API_Service';

jest.mock('../../Services/API_Service');
const mockgetchecktoEval = mocked(API_Service.checktoEval,true)

describe('When evrything is OK in evaluated cheque',()=>{
    beforeEach(async()=>{
        render(<BrowserRouter><Evaluated/></BrowserRouter>)
        await waitFor(()=>expect(mockgetchecktoEval).toHaveBeenCalled())
    })
    test('should render the evaluated cheque Page without crashing',()=>{
        screen.debug()
    })
    test('should render the evaluated cheque Page without crashing',()=>{
        screen.getByText("Cheque ID")
        screen.getByText("Sender ID")
        screen.getByText("Date")
        screen.getByText("PASS")
        screen.getByText("Amount")
    })
})

describe("when the component fethches the details successfully",()=>{
    beforeEach(()=>{
        mockgetchecktoEval.mockClear();
    });
    test('should call evaluated cheque once',async ()=>{
        render(<BrowserRouter><Evaluated/></BrowserRouter>)
        await waitFor(()=>expect(mockgetchecktoEval).toHaveBeenCalledTimes(1))
    })
    test('should display evaluated cheque details correclty',async ()=>{
        mockgetchecktoEval.mockImplementation((callback)=>{
            callback({data:{data:[
            {status:'EVALUATING', cheque_id:'1',sender_id:'23', receiver_id:'3', amount:'5000', date:'2020-02-1', note:'Sample note for mock'},
            // {cheque_id:'19',sender_id:'239', receiver_id:'39', amount:'4000', date:'2020-02-15', note:'Sample note for mock non evaluating'}
        ]}})
          })
    render(<BrowserRouter><Evaluated/></BrowserRouter>)
        screen.getByText('1')
        screen.getByText('23')
        screen.getByText('2020-02-1')
        screen.getByText('5000')
        screen.getByText('Sample note for mock')
    // screen.debug()
    })
    test('should not display - non evaluated',async ()=>{
        mockgetchecktoEval.mockImplementation((callback)=>{
            callback({data:{data:[
            {cheque_id:'19',sender_id:'239', receiver_id:'39', amount:'4000', date:'2020-02-15', note:'Sample note for mock non evaluating'}
        ]}})
          })
        render(<BrowserRouter><Evaluated/></BrowserRouter>)
        expect(screen.queryByText('239')).not.toBeInTheDocument()
        expect(screen.queryByText('4000')).not.toBeInTheDocument()
        expect(screen.queryByText('2020-02-15')).not.toBeInTheDocument()
        expect(screen.queryByText('Sample note for mock non evaluating')).not.toBeInTheDocument()

    // screen.debug()
    })
    test("matches evaluated cheque snapshot",()=>{
        const tree=renderer.create(<Evaluated/>).toJSON()
        expect(tree).toMatchSnapshot();
    })
})