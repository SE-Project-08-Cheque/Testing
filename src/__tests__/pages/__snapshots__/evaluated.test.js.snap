// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the component fethches the details successfully matches evaluated cheque snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": "50px",
      }
    }
  >
    <div
      className="e-card e-card-horizontal"
      style={
        Object {
          "width": "400px",
        }
      }
    >
      <img
        alt="Sample"
        src=""
        style={
          Object {
            "height": "80px",
          }
        }
      />
      <div
        className="e-card-stacked"
      >
        <div
          className="e-card-header"
        >
          <div
            className="e-card-header-caption"
          />
        </div>
        <div
          className="e-card-content"
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Cheque ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Sender ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Receiver ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Amount
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Date
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Note
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            PASS
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            RETURN
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>,
]
`;
